base_layer/core/src/fee.rs

```diff
@@ -31,9 +31,7 @@ pub const BASE_COST: u64 = 1;
 impl Fee {
     /// Computes the absolute transaction fee given the fee-per-gram, and the size of the transaction
     pub fn calculate(fee_per_gram: MicroTari, num_inputs: usize, num_outputs: usize) -> MicroTari {
-        (BASE_COST +
-            (COST_PER_INPUT * num_inputs as u64 + COST_PER_OUTPUT * num_outputs as u64) * u64::from(fee_per_gram))
-        .into()
+        (BASE_COST + Fee::calculate_weight(num_inputs, num_outputs) * u64::from(fee_per_gram)).into()
```

Comment:
I think you can multiply a `MicroTari` by a u64.. maybe the order of operands matters

Reply:
I shuffled things around, but it didn't seem to work without the explicit conversion.